src/keys/slip10.rs

```diff
@@ -253,18 +285,22 @@ impl<K: hazmat::IsSecretKey> From<&Seed> for Slip10<K> {
     }
 }
 
-impl<K: hazmat::IsPublicKey> Slip10<K> {
+impl<K: hazmat::IsPublicKey> Slip10<K>
+where
+    K::SecretKey: hazmat::ToPublic,
+{
```

Comment:
I'ld probably put `K: hasmat::IsPublicKey` under the `where` clause as well, whichs the bounds easier to read IMO.

Reply:
Agree. Done.